tests/examples/aws-java-pulumiup-app/src/main/java/com/example/pulumi/JarDeployment.java

```diff
@@ -0,0 +1,192 @@
+package com.example.pulumi;
+
+import com.pulumi.asset.FileAsset;
+import com.pulumi.aws.s3.BucketObject;
+import com.pulumi.aws.s3.BucketObjectArgs;
+import com.pulumi.core.Output;
+import com.pulumi.kubernetes.Provider;
+import com.pulumi.kubernetes.ProviderArgs;
+import com.pulumi.kubernetes.apps_v1.Deployment;
+import com.pulumi.kubernetes.apps_v1.DeploymentArgs;
+import com.pulumi.kubernetes.apps_v1.inputs.DeploymentSpecArgs;
+import com.pulumi.kubernetes.core_v1.Namespace;
+import com.pulumi.kubernetes.core_v1.NamespaceArgs;
+import com.pulumi.kubernetes.core_v1.Service;
+import com.pulumi.kubernetes.core_v1.ServiceArgs;
+import com.pulumi.kubernetes.core_v1.enums.ServiceSpecType;
+import com.pulumi.kubernetes.core_v1.inputs.ContainerArgs;
+import com.pulumi.kubernetes.core_v1.inputs.ContainerPortArgs;
+import com.pulumi.kubernetes.core_v1.inputs.EmptyDirVolumeSourceArgs;
+import com.pulumi.kubernetes.core_v1.inputs.EnvVarArgs;
+import com.pulumi.kubernetes.core_v1.inputs.PodSpecArgs;
+import com.pulumi.kubernetes.core_v1.inputs.PodTemplateSpecArgs;
+import com.pulumi.kubernetes.core_v1.inputs.ServicePortArgs;
+import com.pulumi.kubernetes.core_v1.inputs.ServiceSpecArgs;
+import com.pulumi.kubernetes.core_v1.inputs.VolumeArgs;
+import com.pulumi.kubernetes.core_v1.inputs.VolumeMountArgs;
+import com.pulumi.kubernetes.meta_v1.inputs.LabelSelectorArgs;
+import com.pulumi.kubernetes.meta_v1.inputs.ObjectMetaArgs;
+import com.pulumi.resources.ComponentResource;
+import com.pulumi.resources.CustomResourceOptions;
+
+import java.util.List;
+import java.util.Map;
+
+public class JarDeployment extends ComponentResource {
+    static final String TARGET_DIR = "/var/run/secrets/java";
+    public final Output<String> endpoint;
+
+    public JarDeployment(String name, LambdaArgs applyArgs) {
+        super("example:spring:JarDeployment",
+                name,
+                JarDeployment.apply(applyArgs),
+                null);
+
+        // Check out JarDeploymentArgs
+        final var args = JarDeployment.apply(applyArgs);
+
+        final var appJar = deployJar(args.getJarPath(), args.getBucketName());
+        this.endpoint = deployApp(args.getKubeconfig(), appJar);
+
+        return;
+    }
+
+    public interface LambdaArgs {
```

Comment:
nit: you can add `@FunctionalInterface` https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html